### PR TITLE
8272398: Update DockerTestUtils.buildJdkDockerImage()

### PR DESCRIPTION
**Pull request by mseledts**
Please review this change that updates the buildJdkDockerImage() test library API.

This work originated while working on "8195809: [TESTBUG] jps and jcmd -l support for containers is not tested".
The initial intent was to extend the buildJdkDockerImage() API of DockerTestUtils to accept custom Dockerfile content.
As I analyzed the usage of buildJdkDockerImage() I realized that:
  - 2nd argument "dockerfile" is always the same: "Dockerfile-BasicTest"
      its use has been obsolete for some time, in favor of Dockerfile generated by DockerTestUtils
  - 3rd argument "buildDirName" is also always the same: "jdk-docker"

Hence I thought it would be a good idea to simplify this API and make it up-to-date.

Also, since the method signature is being updated, I thought it would be a good idea to also change the name to use more generic container terminology:
buildJdkDockerImage() --> buildJdkContainerImage()

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8272398](https://bugs.openjdk.java.net/browse/JDK-8272398): Update DockerTestUtils.buildJdkDockerImage()

### Reviewers
 * [Igor Ignatyev](https://openjdk.java.net/census#iignatyev) (@iignatev - **Reviewer**) ⚠️ Review applies to d211c220f9ebd1ed29d24256936e40df05964bb3
 * [Harold Seigel](https://openjdk.java.net/census#hseigel) (@hseigel - **Reviewer**) ⚠️ Review applies to d211c220f9ebd1ed29d24256936e40df05964bb3

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5134/head:pull/5134` \
`$ git checkout pull/5134`

Update a local copy of the PR: \
`$ git checkout pull/5134` \
`$ git pull https://git.openjdk.java.net/jdk pull/5134/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5134`

View PR using the GUI difftool: \
`$ git pr show -t 5134`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5134.diff">https://git.openjdk.java.net/jdk/pull/5134.diff</a>

</details>

### CODE REVIEW

**Comment by mseledts** [comment]
@hseigel Could you please review when you have a chance ?

**Comment by mseledts** [comment]
Testing:
  - ran jdk/containers and hotspot/containers tests on
     - OL 7.9 with Docker
     - OL 8.3 with Podman
     - OL 8.3 aarch64 with Podman
All PASS

File: test/lib/jdk/test/lib/containers/docker/DockerTestUtils.java [hunk 1]
```diff
@@ -120,59 +120,64 @@ private static boolean isDockerEngineAvailableCheck() throws Exception {
...
      * @throws Exception
      */
-    public static void
-        buildJdkDockerImage(String imageName, String dockerfile, String buildDirName)
-            throws Exception {
+    public static void buildJdkContainerImage(String imageName, String dockerfileContent) throws Exception {
 
-        Path buildDir = Paths.get(".", buildDirName);
+        // Create an image build/staging directory
+        Path buildDir = Paths.get(".", "image-build");
```
Comment (iignatev) [inline test/lib/jdk/test/lib/containers/docker/DockerTestUtils.java:145]: to make it more robust and possible to have more than one container within a test, I'd use `imagName` + "image-build" as build dir name.

Comment (iignatev) [reply test/lib/jdk/test/lib/containers/docker/DockerTestUtils.java:145]: you also don't need to have `.` as a first argument, `Paths.get("x")` returns you a path to "x" in current directory.

File: test/lib/jdk/test/lib/containers/docker/DockerTestUtils.java [hunk 2]
```diff
@@ -120,59 +120,64 @@ private static boolean isDockerEngineAvailableCheck() throws Exception {
...
+     *                   needed to build the image.
      * @throws Exception
      */
-    public static void
-        buildDockerImage(String imageName, Path dockerfile, Path buildDir) throws Exception {
-
-        generateDockerFile(buildDir.resolve("Dockerfile"),
-                           DockerfileConfig.getBaseImageName(),
-                           DockerfileConfig.getBaseImageVersion());
+    public static void buildImage(String imageName, Path buildDir) throws Exception {
```
Comment (iignatev) [inline test/lib/jdk/test/lib/containers/docker/DockerTestUtils.java:179]: it seems there are no users of this method, do we need to keep it public?

**Review by iignatev — CHANGES_REQUESTED** [review]

File: test/lib/jdk/test/lib/containers/docker/DockerTestUtils.java (hunk 2 above)
Comment (mseledts) [reply test/lib/jdk/test/lib/containers/docker/DockerTestUtils.java:179]: Sounds good. Will do.

**Review by iignatev — APPROVED** [review]
Looks good and trivial to me.

File: test/lib/jdk/test/lib/containers/docker/DockerTestUtils.java [hunk 3]
```diff
@@ -120,59 +120,66 @@ private static boolean isDockerEngineAvailableCheck() throws Exception {
...
-        buildDockerImage(imageName, Paths.get(Utils.TEST_SRC, dockerfile), buildDir);
+
+        buildImage(imageName, buildDir);
     }
 
 
     /**
-     * Build a docker image based on given docker file and docker build directory.
+     * Build a container image based on given Dockerfile and image build directory.
      *
```
Comment (hseigel) [inline test/lib/jdk/test/lib/containers/docker/DockerTestUtils.java:175]: This comment looks wrong because there is no longer a given Dockerfile.

**Review by hseigel — APPROVED** [review]
Other than that one obsolete comment the changes look good.
Thanks, Harold

File: test/lib/jdk/test/lib/containers/docker/DockerTestUtils.java (hunk 3 above)
Comment (mseledts) [reply test/lib/jdk/test/lib/containers/docker/DockerTestUtils.java:175]: Thanks. I will remove this comment.

**Comment by mseledts** [comment]
Igor, Harold, thank you for review.

/integrate

**Pull request closed without merging**